src/table/interfaces.tsx

```diff
@@ -321,25 +321,20 @@ export namespace TableProps {
     validation?: (item: T, value: Optional<V>) => Optional<string>;
   }
 
-  export type ColumnEditConfig<ItemType, ValueType> =
-    | {
-        editConfig?: undefined;
-        cell: CellRenderer<ItemType>;
-      }
-    | {
-        editConfig: EditConfig<ItemType, ValueType>;
-        cell: EditableCellRenderer<ItemType, ValueType>;
-      };
-
-  export type ColumnDefinition<ItemType, ValueType = any> = {
+  export interface ColumnEditConfig<ItemType> {
+    editConfig?: EditConfig<ItemType>;
+    cell(item: ItemType, context?: CellContext<any>): React.ReactNode;
```

Comment:
I'll update the example (editable-with-app-layout) to use correct types, but I'm not sure there's much we can do to ensure type consistency across the two fields.
I'm open to suggestions if you think theres a way to prevent the scenario you described from happening.

Reply:
Following our discussion offline, I've updated the types for the submit function to use `any` for the type for `newValue`.